Data/Aeson/Parser/Internal.hs

```diff
@@ -390,12 +391,18 @@ eitherDecodeWith p to s =
       L.Done _ v     -> case to v of
                           ISuccess a      -> Right a
                           IError path msg -> Left (path, msg)
-      L.Fail _ ctx msg -> Left ([], buildMsg ctx msg)
+      L.Fail notparsed ctx msg -> Left ([], buildMsg notparsed ctx msg)
   where
-    buildMsg :: [String] -> String -> String
-    buildMsg [] msg = msg
-    buildMsg (expectation:_) msg =
-      msg ++ ". Expecting " ++ expectation
+    buildMsg :: L.ByteString -> [String] -> String -> String
+    buildMsg notParsed [] msg = msg ++ grabErrorLine notParsed
+    buildMsg notparsed (expectation:_) msg =
+      msg ++ ". Expecting " ++ expectation ++ grabErrorLine notparsed
+    grabErrorLine :: L.ByteString -> String
+    grabErrorLine bs =
+        let notParsedStr = C.unpack bs
```

Comment:
unpacking 10MB input is also not a good idea.

Reply:
Would you be happy with more performant solution? or you want to drop the idea completely?

Reply:
Something which
1. [`take`](https://hackage.haskell.org/package/bytestring-0.10.10.0/docs/Data-ByteString-Lazy.html#v:take)s say ~100 characters
2. then after [`takeWhile`](https://hackage.haskell.org/package/bytestring-0.10.10.0/docs/Data-ByteString-Lazy.html#v:takeWhile) `/= 10` (for `'\n'`) and maybe `dropWhile (\w8 -> w8 == 32)` (space, you may want to add tab `9` and CR `13`), i.e. taking single line and dropping whitespace
3. and finally converts that small lazy `ByteString` to `String` for display should be fine

Note: I'm not a maintainer though, it's up to @bergmark 

I also wouldn't try to reformat the input any further, the literal part works quite well in my experience (I use something similar in other parsers). You'll get something like

```
Error in $: Failed reading: satisfy. Expecting object value at '"postal_code": "14236"'
```

Reply:
Nice, let's see what @bergmark has to say about this and if the light is green we continue.

Reply:
@phadej is as much a maintainer, just permission issues on github T_T. (@phadej if you want to be added to maintainers in .cabal and hackage LMK! I thought I already added you on hackage...)

What will you get from this if your input is only one line? Will it start after the error occurs or show the beginning of the line? I think I usually use compacted JSON, even for testing, but I might be an outlier here!

Sounds good overall

Reply:
@phadej I just pushed some changes related to performance. 

> What will you get from this if your input is only one line?  Will it start after the error occurs or show the beginning of the line?

@bergmark It will show the unparsed portion of the bytestring so you should have easier time spotting where exactly the error occured. The newline is not so relevant anymore since we are taking the first 100 characters to show that ` at blah blah`. If error message turns out to be empty this will default to the message we previously had in here.

Reply:
I still want to test this a bit after the latest changes.

Reply:
Ok guys, I tested this a bit and it seems like it is working fine. If you are happy with the changes we can move on. I think this is just a step forward, perhaps we could utilize a proper diffing algorithm using the input bytestring and the parsed portion so we get even nicer output.